src/shady-css/parser.js

```diff
@@ -225,22 +229,21 @@ class Parser {
       }
     }
 
-    rule = tokenizer.slice(ruleStart, ruleEnd);
-
     // A ruleset never contains or ends with a semi-colon.
     if (tokenizer.currentToken.is(Token.type.propertyBoundary)) {
-      let colonIndex = rule.indexOf(':');
+      let declarationName = tokenizer.slice(ruleStart, colon.previous);
```

Comment:
Why do you need to call slice()? is the name ever going to span multiple tokens? One of the advantages of using a linked list of tokens is that you should be able to get rid of slice entirely - since each AST node has a reference to the start and end token you can always create a substring later when needed.